spec-tests/src/test/java/org/candlepin/spec/OwnerProductResourceSpecTest.java

```diff
@@ -0,0 +1,799 @@
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createProductWithProvidedAndDerivedProduct(owner.getKey());
+        assertBadRequest(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), product.getId()));
+    }
+
+    @Test
+    @OnlyInHosted
+    @DisplayName("should return forbidden request on attempt to delete product attached to sub")
+    public void shouldReturnForbiddenRequestOnAttemptToDeleteProductAttachedToSub() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createHostedProductWithProvidedAndDerivedProduct();
+        createUpstreamSubscription(owner, product);
+        assertForbidden(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), product.getId()));
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should return bad request on attempt to delete provided product attached to sub")
+    public void shouldReturnBadRequestOnAttemptToDeleteProvidedProductAttachedToSub() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createProductWithProvidedAndDerivedProduct(owner.getKey());
+        ProductDTO providedProduct = Iterables.getOnlyElement(product.getProvidedProducts());
+        assertBadRequest(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), providedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInHosted
+    @DisplayName("should return forbidden request on attempt to delete provided product attached to sub")
+    public void shouldReturnForbiddenRequestOnAttemptToDeleteProvidedProductAttachedToSub()
+        throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createHostedProductWithProvidedAndDerivedProduct();
+        ProductDTO providedProduct = Iterables.getOnlyElement(product.getProvidedProducts());
+        createUpstreamSubscription(owner, product);
+        assertForbidden(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), providedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInStandalone
+    @DisplayName("should return bad request on attempt to delete derived product attached to sub")
+    public void shouldReturnBadRequestOnAttemptToDeleteDerivedProductAttachedToSub() throws Exception {
+        OwnerDTO owner = ownerApi.createOwner(Owners.random());
+        ProductDTO product = createProductWithProvidedAndDerivedProduct(owner.getKey());
+        ProductDTO derivedProduct = product.getDerivedProduct();
+        assertBadRequest(() -> ownerProductApi.deleteProductByOwner(owner.getKey(), derivedProduct.getId()));
+    }
+
+    @Test
+    @OnlyInHosted
+    @DisplayName("should return forbidden request on attempt to delete derived product attached to sub")
+    public void shouldReturnForbiddenRequestOnAttemptToDeleteDerivedProductAttachedToSub()
```

Comment:
This test can be generalized with custom products and pools -- no need for the hosted test API restriction.

Reply:
Updated

Reply:
Removed